pandas/tests/copy_view/test_constructors.py

```diff
@@ -73,3 +79,37 @@ def test_series_from_series_with_reindex(using_copy_on_write):
     assert not np.shares_memory(ser.values, result.values)
     if using_copy_on_write:
         assert result._mgr.refs is None or result._mgr.refs[0] is None
+
+
+@pytest.mark.parametrize("dtype", [None, "int64"])
+@pytest.mark.parametrize("columns", [None, ["a", "b"], ["a", "b", "c"]])
+def test_dataframe_from_dict_of_series(using_copy_on_write, columns, dtype):
+    # Case: constructing a DataFrame from Series objects with copy=False
+    # has to do a lazy following CoW rules
+    # (the default for DataFrame(dict) is still to copy to ensure consolidation)
+    s1 = Series([1, 2, 3])
+    s2 = Series([4, 5, 6])
```

Comment:
Can you add a test where we have to reindex the Series, e.g. make a copy to check that no copy is triggered when updating this column?

Reply:
I am adding the index as parametrization to the current test (for a reindex case that can give a view), which will already test that we certainly follow CoW semantics. But it of doesn't test that a copy is _not_ triggered. 
I think in general we don't really have tests for that yet? What would be the best way to test this? With some pytest monkeypatch? (but we don't have a single method in the internals that performs the copy on write, so that's probably not that easy)

Reply:
> What would be the best way to test this? 

Or, I suppose I could just get the underlying array before and after mutating the dataframe, and then ensure those values are identical (which guarantees we didn't copy when mutating)

Reply:
Yeah that is probably the best approach, I am using this as well in a couple of places